### DISCUSSION TITLE
[Enhancement] [image-viewer] el-image组件图片预览

### DISCUSSION THREAD

**Discussion by Onion-L** [element-plus/element-plus#18630] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `image-viewer`

### 特性
1. 在图片预览界面添加下载按钮。
2. 点击下载按钮可以将当前查看的图片保存到本地。
3. 支持多种常见图片格式的下载（如 JPG、PNG、GIF 等）。
4. 下载时可以保留原始文件名。

### Additional comments

_(empty)_

**Comment by HaceraI** [comment]
我觉得提供两个 slot (prefix-action、after-action) 更合适，这样不仅仅局限于 下载，有时候某些图片可能也需要提供别的操作（例如打印）

**Reply by Onion-L** [reply]
确实，因为我的应用场景只用到了下载，当然如果能开放查看器的slot就更好了😄

**Comment by zzjiaxiang** [comment]
#19497
